platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
> I also note we're using int extensively for errno values rather than errno_t. If we want poll_t, we should probably change that, together with adding filehandle_t.

I'm ok with this : )

But you're right it's not the case currently, so I'm fine with moving forward as is with the `short` type.

> My current thought on lack of true poll for some network stacks is that we make it report with an honest EINVAL if it doesn't work - fully-portable network applications considering using poll() would have to cope with and expect the EINVAL it would return, and have a backup plan (until the day they can rely on it being available on all stacks they care about). There would be a method in FileHandle to indicate the unavailablility.

Your response makes general sense to me. I agree with going for the straight-forward thing here.